app/routes.py

```diff
@@ -1,2 +1,186 @@
+    headers = {'Authorization': f"Bearer {SLACK_TOKEN}"}
+    requests.post(slack_path, params=query_params, headers=headers)
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def handle_task_completion(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("Not Found", 404)
+    elif request.method == "PATCH":
+        if bool(task.completed_at) == False:
+            task.completed_at = datetime.datetime.now()
+            slack_message =  f"Someone just completed the task {task.title}"
+            post_to_slack(slack_message)
+        return (
+            { "task":{
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": bool(task.completed_at)
+        }}, 200
+        )
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def handle_task_not_completion(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("Not Found", 404)
+    elif request.method == "PATCH":
+        task.completed_at = None
+        return ({ "task":{
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": False
+        }}, 200)
+
+@goals_bp.route("", methods=["GET","POST"])
+def handle_goals():
+    if request.method == "GET":
+        goals = Goal.query.all()
+        goals_response = [] 
+        for goal in goals:   
+            goals_response.append({
+                "id": goal.goal_id,
+                "title": goal.title,
+            })
+        return jsonify(goals_response)
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body:
+            return {"details": f"Invalid data"}, 400
+        new_goal = Goal(
+            title=request_body["title"]
+            )
+        db.session.add(new_goal)
+        db.session.commit()
+
+        return make_response( 
+            {"goal": {
+            "id": new_goal.goal_id,
+            "title": new_goal.title
+        }}, 201
+        )
+
+@goals_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"])
+def handle_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+
+    if goal is None:
+        return make_response("Not Found", 404)
+    elif request.method == "GET":
+        return { "goal":{
+            "id": goal.goal_id,
+            "title": goal.title,
+        }}
+    elif request.method == "PUT":
+        form_data = request.get_json()
+
+        goal.title = form_data["title"]
+
+        db.session.commit()
+
+        return {
+            "goal":{
+            "id": goal.goal_id,
+            "title": goal.title
+        }}
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+        return make_response({
+            "details":f"Goal {goal.goal_id} \"{goal.title}\" successfully deleted"
+        })
```

Comment:
To get wave six passing go ahead and follow what you've been doing! 
Set up the route `@goal_bp.route("/<goal_id>/tasks", methods=["GET", "POST"])` and implement what I put in the comments for the Models.